### PR TITLE
Data processing and loader

### PR DESCRIPTION
**Pull request by ggantos**

### CODE REVIEW

File: ptype/data.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        source (str): Precip observation source. Supports 'ASOS' or 'mPING'.
+        train_val_test_proportion (list of int): proportion of data to use in each train, validation, and test split.
+        seed (int): random seed used in random shuffle.
+    Returns:
+    Dictionary of Pandas dataframes of training / validation / test data
+    """
+    
+    dates = [x[-16:-8] for x in os.listdir(os.path.join(data_path, source))]
+    random.Random(seed).shuffle(dates)
+
```
Comment (charlie-becker) [inline ptype/data.py]: I think this is a scenario where we don't want completely random shuffles between the different splits due to temporal contamination. An easy fix with the current code would be to sort the dates and remove the shuffle.

`dates = sorted([x[-16:-8] for x in os.listdir(os.path.join(data_path, source))])`

However, you might also consider using a date sorter so you can specific specific ranges for each split. That would probably require running the function for each split using a `start_date` and `end_date` parameter and then converting them to strings with something like 

`pd.date_range(start_date, end_date, freq='h').strftime("%Y%m%d%")`

File: ptype/data.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        scalers["output"].fit_transform(data["train"][output_features].values.reshape(-1, len([output_features]))),
+        columns=[output_features])
+    scaled_data["val_y"] = pd.DataFrame(
+        scalers["output"].transform(data["val"][output_features].values.reshape(-1, len([output_features]))),
+        columns=[output_features])
+    scaled_data["test_y"] = pd.DataFrame(
+        scalers["output"].transform(data["test"][output_features].values.reshape(-1, len([output_features]))),
+        columns=[output_features])
+    
+    return scaled_data, scalers
```
Comment (charlie-becker) [inline ptype/data.py]: Since our output is categorical, we shouldn't need an output scaler but would need the output encoded for a categorical model.

**Review by charlie-becker — CHANGES_REQUESTED** [review]
Thanks for getting together so quickly! I added a couple of comments regarding non-shuffled data splitting and preprocessing for categorical output.

Edit: One more thing - there is something buggy with the base configuration of the github actions CI workflow from the ai2es template that prevents it from succesfully completing (and sometimes runs for hours before failing). I used a slightly different setup in [(https://github.com/ai2es/XAI-quant)] (more similar to HWT, etc) that seems to work. Not sure this is a priority, but I'll bring it up at out planning meeting.

File: ptype/data.py (hunk 1 above)
Comment (ggantos) [reply ptype/data.py]: Thanks Charlie - I thought that temporal contamination was important to worry about within the same day but not among days within the same year/month/season? In any case, I've updated the code to reflect start and end dates.

File: ptype/data.py (hunk 2 above)
Comment (ggantos) [reply ptype/data.py]: Added options for one-hot and label encoding.

**Review by djgagne — APPROVED** [review]
